tuned/plugins/plugin_service.py

```diff
@@ -0,0 +1,275 @@
+
+	def disable(self, name, runlevel):
+		cmd.execute(["sysv-rc-conf", "--level", runlevel, name, "off"])
+
+	def is_enabled(self, name, runlevel):
+		(retcode, out) = cmd.execute(["sysv-rc-conf", "--list", name])
+		return out.split("%s:" % str(runlevel))[1][:2] == "on" if retcode == 0 else None
+
+class OpenRCHandler(InitHandler):
+	def runlevel_get(self):
+		(retcode, out) = cmd.execute(["rc-status", "-r"])
+		return out.strip() if retcode == 0 else None
+
+	def start(self, name):
+		cmd.execute(["rc-service", name, "start"])
+
+	def stop(self, name):
+		cmd.execute(["rc-service", name, "stop"])
+
+	def enable(self, name, runlevel):
+		cmd.execute(["rc-update", "add", name, runlevel])
+
+	def disable(self, name, runlevel):
+		cmd.execute(["rc-update", "del", name, runlevel])
+
+	def is_running(self, name):
+		(retcode, out) = cmd.execute(["rc-service", name, "status"], no_errors = [0])
+		return retcode == 0
+
+	def is_enabled(self, name, runlevel):
+		(retcode, out) = cmd.execute(["rc-update", "show", runlevel])
+		return bool(re.search(r"\b" + re.escape(name) + r"\b", out))
+
+class SystemdHandler(InitHandler):
+	# runlevel not used
+	def runlevel_get(self):
+		return ""
+
+	def start(self, name):
+		cmd.execute(["systemctl", "restart", name])
+
+	def stop(self, name):
+		cmd.execute(["systemctl", "stop", name])
+
+	def enable(self, name, runlevel):
+		cmd.execute(["systemctl", "enable", name])
+
+	def disable(self, name, runlevel):
+		cmd.execute(["systemctl", "disable", name])
+
+	def is_running(self, name):
+		(retcode, out) = cmd.execute(["systemctl", "is-active", name], no_errors = [0])
+		return retcode == 0
+
+	def is_enabled(self, name, runlevel):
+		(retcode, out) = cmd.execute(["systemctl", "is-enabled", name], no_errors = [0])
+		return retcode == 0
+
+	def cfg_install(self, name, cfg_file):
+		if not os.path.exists(cfg_file):
+			log.error("Unable to find service configuration '%s'" % cfg_file)
+			return
+		dirpath = consts.SERVICE_SYSTEMD_CFG_PATH % name
+		try:
+			os.makedirs(dirpath, consts.DEF_SERVICE_CFG_DIR_MODE)
+		except OSError as e:
+			log.error("Unable to create directory '%s': %s" % (dirpath, e))
+			return
+		cmd.copy(cfg_file, dirpath)
+
+	def cfg_uninstall(self, name, cfg_file):
+		dirpath = consts.SERVICE_SYSTEMD_CFG_PATH % name
+		path = "%s/%s" % (dirpath, os.path.basename(cfg_file))
+		cmd.unlink(path)
+		# remove the service dir if empty, do not check for errors
+		try:
+			os.rmdir(dirpath)
+		except (FileNotFoundError, OSError):
+			pass
+
+	def cfg_verify(self, name, cfg_file):
+		if cfg_file is None:
+			return None
+		path = "%s/%s" % (consts.SERVICE_SYSTEMD_CFG_PATH % name, os.path.basename(cfg_file))
+		if not os.path.exists(cfg_file):
+			log.error("Unable to find service '%s' configuration '%s'" % (name, cfg_file))
+			return False
+		if not os.path.exists(path):
+			log.error("Service '%s' configuration not installed in '%s'" % (name, path))
+			return False
+		md5sum1 = cmd.md5sum(cfg_file)
```

Comment:
I am a bit unsure about this. MD5?

Reply:
Yep, it's very simple check that the file wasn't modified, it's unlikely that somebody will play with the md5 collisions on the service configuration overlay file just for the profile verification to fail :D. But no problem to change to e.g. sha-256 or whatever is supported by hashlib.